backend/database/migrations/create-posts.js

```diff
@@ -1,19 +1,38 @@
-'use strict';
-module.exports = {
-    up: (queryInterface, Sequelize) => {
-        return queryInterface.createTable('posts', {
-            id: {
-                allowNull: false,
-                autoIncrement: true,
-                primaryKey: true,
-                type: Sequelize.INTEGER
-            },
-            text_content: {
-                type: Sequelize.STRING
-            }
-        });
-    },
-    down: (queryInterface, Sequelize) => {
-        return queryInterface.dropTable('posts');
-    }
-};
+module.exports = {
+    up: (queryInterface, Sequelize) =>
+        queryInterface.createTable('posts', {
+            id: {
+                allowNull: false,
+                autoIncrement: true,
+                primaryKey: true,
+                type: Sequelize.INTEGER,
+            },
+            text_content: {
+                type: Sequelize.STRING,
+            },
+            author: {
+                type: Sequelize.STRING,
+            },
+            parent: {
+                type: Sequelize.INTEGER,
+            },
+            usersLiked: {
+                type: Sequelize.INTEGER,
+            },
+            usersShared: {
+                type: Sequelize.INTEGER,
+            },
+            createdAt: {
+                allowNull: false,
+                type: Sequelize.DATE,
+            },
+            updatedAt: {
+                allowNull: false,
+                type: Sequelize.DATE,
+            },
+            attachments: {
+                type: Sequelize.STRING,
+            },
+        }),
```

Comment:
The actual model and the response we send to FE is slightly different, the real model, the one we save to the DB should match this:
**Post**
* id
* content (string)
* author (userID)
* parent (ID of parent post)
* children (array of post IDs)
* usersLiked (array of IDs of users who likes the post)
* usersShared (array of IDs users who shared the post)
* timestamp (unix timestamp indicating timestamp of creation)
* Attachments (link to attachment in Firebase db)

After you do all the processing in your controllers, you should convert the object you're about to send as a response to this format:
![image](https://user-images.githubusercontent.com/68842181/157351472-84962b45-0c8e-45e2-a91f-64de0aae5f7e.png)

Reply:
You might need to create a postsDTO class and have a method that creates a postsDTO object from the whole post object

Reply:
Right now attachments is a seperate issue, so I don't think I need to fix that yet. 
As for usersLiked and usersShared, I'm going to leave those as integers for testing purposes (at least for now).

Reply:
I fixed the author field though.

Reply:
Agreed, posts models can be finalised in a separate issue, for now we just need the CRUD endpoints